src/wallet/sqlite.cpp

```diff
@@ -374,16 +374,41 @@ bool SQLiteBatch::HasKey(CDataStream&& key)
 
 bool SQLiteBatch::StartCursor()
 {
-    return false;
+    assert(!m_cursor_init);
+    if (!m_database.m_db) return false;
+    m_cursor_init = true;
+    return true;
 }
 
 bool SQLiteBatch::ReadAtCursor(CDataStream& key, CDataStream& value, bool& complete)
 {
-    return false;
+    complete = false;
+
+    if (!m_cursor_init) return false;
+
+    int res = sqlite3_step(m_cursor_stmt);
+    if (res == SQLITE_DONE) {
+        complete = true;
+        return true;
+    } else if (res != SQLITE_ROW) {
+        LogPrintf("SQLiteBatch::ReadAtCursor: Unable to execute cursor step: %s\n", sqlite3_errstr(res));
+        return false;
+    }
+
+    // Leftmost column in result is index 0
+    const char* key_data = (const char*)sqlite3_column_blob(m_cursor_stmt, 0);
```

Comment:
293bdfcdc1825e1b288ee09584cf26979a4d5900
```suggestion
    const char* key_data = reinterpret_cast<const char*>(sqlite3_column_blob(m_cursor_stmt, 0));
```

Reply:
Done